src/main/webapp/script.js

```diff
@@ -90,17 +99,55 @@ function drawRegionsMap(censusDataArray, title) {
 
 // createDataArray takes in the data array returned by the census API
 // and reformats it into a data table for the visualization API.
-function createDataArray(censusDataArray) {
+function createDataArray(censusDataArray, description) {
   const vizDataArray = [];
-  censusDataArray.forEach((state) => {
-    vizDataArray.push([state[0], state[1]/1000]);
-  });
-  // Changes the header of the vizDataArray to match the Visualization API
+  // Check to see if an extra calculation for percentages is needed
+  if (checkPercentage(censusDataArray[0])) {
+    censusDataArray.forEach((state) => {
+      vizDataArray.push([state[0], percentToTotal(state[1], state[2])]);
+    });
+  } else {
+    censusDataArray.forEach((state) => {
+      vizDataArray.push([state[0], state[1]/1000]);
+    });
+  }
+  // Changes the header of the vizDataArray to match Visualization API
   vizDataArray[0][0] = 'State';
-  vizDataArray[0][1] = 'Population';
+  // Add a more accurate descriptor for each state
+  vizDataArray[0][1] = 'Population of people ' +
+    description.age + ' who ' + description.action;
+  // make it moved to
+  if (description.action === 'moved') {
+    vizDataArray[0][1] += ' to';
+  }
   return vizDataArray;
 }
 
+// percentToTotal takes in the total number of people in a category
```

Comment:
This is ok for now but I think the cleaner solution is to have the backend put the headers of the form ('population', 'percentage' etc.) and the front does not need to know about things like 'S0201_157E', 'S0201_126E'

Reply:
Got it. The three of us can discuss this soon.